cms/djangoapps/contentstore/views/tabs.py

```diff
@@ -173,7 +156,7 @@ def create_new_list(requested_tab_id_locators, old_tab_list):
     # global or course settings.  so add those to the end of the list.
     non_displayed_tabs = set(old_tab_list) - set(new_tab_list)
     new_tab_list.extend(non_displayed_tabs)
-    return new_tab_list
+    return sorted(new_tab_list, key=lambda item: item.priority or float('inf'))
```

Comment:
I've realised that the simplification I've made here won't work since they rely on the user providing all tab ids. I'll revert some of these changes.